src/app/clusters/laundry-dryer-controls-server/laundry-dryer-controls-server.cpp

```diff
@@ -0,0 +1,186 @@
+{
+    if (aPath.mClusterId != LaundryDryerControls::Id)
+    {
+        // We shouldn't have been called at all.
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    switch (aPath.mAttributeId)
+    {
+    case Attributes::SupportedDrynessLevels::Id:
+        return ReadSupportedDrynessLevels(aPath, aEncoder);
+    default:
+        break;
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR LaundryDryerControlsServer::ReadSupportedDrynessLevels(const ConcreteReadAttributePath & aPath,
+                                                                  AttributeValueEncoder & aEncoder)
+{
+    Delegate * delegate = GetDelegate(aPath.mEndpointId);
+    VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Delegate is nullptr"));
+
+    return aEncoder.EncodeList([delegate](const auto & encoder) -> CHIP_ERROR {
+        for (uint8_t i = 0; true; i++)
+        {
+            DrynessLevelEnum supportedDrynessLevel;
+            auto err = delegate->GetSupportedDrynessLevelAtIndex(i, supportedDrynessLevel);
+            if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+            {
+                return CHIP_NO_ERROR;
+            }
+            ReturnErrorOnFailure(err);
+            ReturnErrorOnFailure(encoder.Encode(supportedDrynessLevel));
+        }
+    });
+}
+
+/**********************************************************
+ * Register LaundryDryerControlsServer
+ *********************************************************/
+
+void MatterLaundryDryerControlsPluginServerInitCallback()
+{
+    LaundryDryerControlsServer & laundryDryerControlsServer = LaundryDryerControlsServer::Instance();
+    registerAttributeAccessOverride(&laundryDryerControlsServer);
+}
+
+Status MatterLaundryDryerControlsClusterServerPreAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath,
+                                                                          EmberAfAttributeType attributeType, uint16_t size,
+                                                                          uint8_t * value)
+{
+    Delegate * delegate = GetDelegate(attributePath.mEndpointId);
+    VerifyOrDie((delegate != nullptr) && "Dryer Controls implementation requires a registered delegate for validation.");
+    switch (attributePath.mAttributeId)
+    {
+    case Attributes::SelectedDrynessLevel::Id: {
+        uint8_t drynessLevelIdx = 0;
+        while (true)
+        {
+            DrynessLevelEnum supportedDryness;
+            auto err = delegate->GetSupportedDrynessLevelAtIndex(drynessLevelIdx, supportedDryness);
+            if (err != CHIP_NO_ERROR)
+            {
+                // Can't find the attribute to be written in the supported list (CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+                // Or can't get the correct supported list
+                return Status::InvalidInState;
+            }
+            if (supportedDryness == static_cast<DrynessLevelEnum>(*value))
```

Comment:
Should statically assert that sizeof(DrynessLevelEnum) == sizeof(*value) or something like that to make this safe.

Reply:
Added the static assert for the DrynessLevelEnum